### PR TITLE
Refactor ephemeral status changes

### PR DESCRIPTION
**Pull request by hausdorff**
> Implement status sinks 
>
> This commit reverts most of #1853 and replaces it with functionally
> identical logic, using the notion of status message-specific sinks.
>
> In other words, where the original commit implemented ephemeral status
> messages by adding an `isStatus` parameter to most of the logging
> methdos in pulumi/pulumi, this implements ephemeral status messages as a
> parallel logging sink, which emits _only_ ephemeral status messages.
>
> The original commit message in that PR was:
>
> > Allow log events to be marked "status" events
> >
> > This commit will introduce a field, IsStatus to LogRequest. A "status"
> > logging event will be displayed in the Info column of the main
> > display, but will not be printed out at the end, when resource
> > operations complete.
> >
> > For example, for complex resource initialization, we'd like to display
> > a series of intermediate results: [1/4] Service object created, for
> > example. We'd like these to appear in the Info column, but not at the
> > end, where they are not helpful to the user.

### CODE REVIEW

File: pkg/engine/events.go [hunk 1]
```diff
@@ -55,13 +55,13 @@ func cancelEvent() Event {
...
-	Severity diag.Severity
-	StreamID int32
-	IsStatus bool
+	URN           resource.URN
+	Prefix        string
+	Message       string
+	Color         colors.Colorization
+	Severity      diag.Severity
+	StreamID      int32
+	StatusMessage bool
```
Comment (CyrusNajmabadi) [inline pkg/engine/events.go]: i do not like this name.  esp. since 'Message' is a string :)   i think 'Status' is fine as a name.

**Comment by hausdorff** [comment]
@CyrusNajmabadi Let's go with `Ephemeral` instead of `StatusMessage` or `IsStatus`

File: pkg/resource/deploy/deploytest/pluginhost.go [hunk 2]
```diff
@@ -91,8 +93,12 @@ func (host *pluginHost) ServerAddr() string {
...
 func (host *pluginHost) Log(sev diag.Severity, urn resource.URN, msg string, streamID int32,
-	isStatus bool) {
-	host.sink.Logf(sev, diag.StreamMessage(urn, msg, streamID), isStatus)
+	ephemeral bool) {
+	if ephemeral {
+		host.statusSink.Logf(sev, diag.StreamMessage(urn, msg, streamID))
+	} else {
+		host.sink.Logf(sev, diag.StreamMessage(urn, msg, streamID))
+	}
 }
```
Comment (lukehoban) [inline pkg/resource/deploy/deploytest/pluginhost.go]: Any reason not to add a `LogStatus` to this API that does the `statusSink` version?  That avoids forcing `, false` into a ton of places inside providers.

Comment (hausdorff) [reply pkg/resource/deploy/deploytest/pluginhost.go]: I don't understand this suggestion. You'd still have to add `, &LogStatus{}`, no?

File: pkg/engine/events.go (hunk 1 above)
Comment (hausdorff) [reply pkg/engine/events.go]: Changed to `Ephemeral`.

File: pkg/resource/deploy/deploytest/pluginhost.go (hunk 2 above)
Comment (lukehoban) [reply pkg/resource/deploy/deploytest/pluginhost.go]: I mean two methods:

```go
host.Log()
host.LogStatus()
```

Comment (joeduffy) [reply pkg/resource/deploy/deploytest/pluginhost.go]: I'll admit, I wondered the same.

**Pull request merged**